jetty-io/src/main/java/org/eclipse/jetty/io/ChannelEndPoint.java

```diff
@@ -357,12 +353,16 @@ public Runnable onSelected()
         return task;
     }
 
+    private void updateKey(Selector selector)
+    {
+        _selector.runKeyAction(selector, _channel, _key, this::updateKey);
```

Comment:
you may as well return the key here:
```suggestion
        _key = _selector.runKeyAction(selector, _channel, _key, this::updateKey);
```

Reply:
I don't think it's a good idea to return the key, since it's the `Consumer` that is supposed to act on the key so there is no point returning it to the caller.

Reply:
let's get rid of the `_key` field (as above)